### PR TITLE
Allow scalar Nengo objects in action rules

### PR DESCRIPTION
**Pull request by jgosmann**
**Motivation and context:**
Allowing standard Nengo objects for scalars in action rules simplifies writing these rules, especially when routed by the action selection system. (Otherwise connections and gates have to be created manually.)

**Interactions with other PRs:**
none

**How has this been tested?**
added a test

**How long should this take to review?**

- Quick (less than 40 lines changed or changes are straightforward)

**Types of changes:**

- New feature (non-breaking change which adds functionality)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [x] I have included a changelog entry.
- [x] I have added tests to cover my changes.
- [x] I have run the test suite locally and all tests passed.

### CODE REVIEW

**Comment by jgosmann** [comment]
Not sure if there is any place in the documentation where this feature should be mentioned. I don't think we are explaining the syntax that detailed anywhere ...

**Comment by Seanny123** [comment]
The one place the syntax comes up is the custom module example, which is what caused me to open #139 in the first place. Consequently, I updated the example to reflect this new capability, but I don't think it needs to be highlighted anywhere else, other than a changelog entry.

File: docs/examples/custom_module.ipynb [hunk 1]
```diff
@@ -122,6 +122,8 @@
       "    model.config[GatedMemory].neurons_per_dimension = 150\n",
       "    \n",
       "    spa_in = spa.Transcode(\"OKAY\", output_vocab=dimensions)\n",
+      "    gate_in = nengo.Node(lambda t: 1 if t < 0.1 else 0)\n",
```
Comment (jgosmann) [inline docs/examples/custom_module.ipynb]: Is this used anywhere?

Comment (Seanny123) [reply docs/examples/custom_module.ipynb]: Oops. Now it is.

**Pull request merged**